src/pow.rs

```diff
@@ -381,218 +322,227 @@ fn count_leading_zeros(data: &[u8]) -> u8 {
     leading_zeros
 }
 
-/// The faucet will dynamically adjust the difficulty based on the
-/// current balance of the faucet to make it increasingly difficult
-/// to retrieve funds from the faucet. The actual equation for this
-/// is:
-///
-/// ```math
-/// f_unclamped(x)=(M-m)(1-\log_{q}\frac{x}{b})+m
-/// f_clamped(x) = \max(m,\min(M,f_unclamped(x)))
-/// f(x) = f_clamped(x)
-/// ```
-///
-/// where:
-///
-/// - `x` is the current balance in BTC
-/// - `M` is the maximum difficulty
-/// - `m` is the minimum difficulty
-/// - `b` is the minimum balance in BTC
-/// - `q` is the amount emitted per request in BTC
-///
-/// # Guarantees
-///
-/// This function guarantees that the difficulty will be between `min_difficulty`
-/// and `max_difficulty` given that the correctness assumptions are met.
-///
-/// # Correctness
-///
-/// For this function correctly output, you must ensure:
-///
-/// - `per_emission` > `Amount::ONE_BTC`, ideally >2 BTC due to the way the curve
-///   functions
-/// - `min_difficulty` <= `max_difficulty`
-/// - `max_difficulty`, `min_difficulty`, `min_balance`, `q` and `x` all > 0
-///
-/// # Expected clamping points
-///
-/// f(x) = M at x = b
-/// f(x) = m at x = bq.
-pub fn calculate_difficulty(x: f32, big_m: f32, m: f32, b: f32, q: f32) -> f32 {
-    // optimisation for when the balance is less than or equal to the min balance
-    if x <= b {
-        return big_m;
+pub struct DifficultyConfig {
+    big_m: u8,
+    m: u8,
+    b: f32,
+    /// Optimisation for when x >= b+Lq, which should be the majority of the time
+    min_diff_start: f32,
+    /// Optimisation for the linear function. This is the gradient of the linear function.
+    precompute_big_a: f32,
+    /// Optimisation for the linear function. This is the y-intercept of the linear function.
```

Comment:
in the `pow.md` We have 1 mention of "American" optimization and 1 "British" optimisation. I prefer we keep American and suggested that in the `pow.md`. Hence, here:


```suggestion
    /// Optimization for when x >= b+Lq, which should be the majority of the time
    min_diff_start: f32,
    /// Optimization for the linear function. This is the gradient of the linear function.
    precompute_big_a: f32,
    /// Optimization for the linear function. This is the y-intercept of the linear function.
```